Tasks/Maven/maven.ps1

```diff
@@ -28,14 +32,37 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.TestResults"
 
-if($jdkVersion -and $jdkVersion -ne "default")
+# If JAVA_HOME is being set by choosing a JDK version find the path to that specified version else use the path given by the user
+$jdkPath = $null
+if($javaHomeSelection -eq 'JDKVersion')
 {
-    $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
-    if (!$jdkPath) 
+    Write-Verbose "Using JDK version to find and set JAVA_HOME"
+    # If the JDK version is not the deafult set the jdkPath to the new JDK version selected
+    if($jdkVersion -and $jdkVersion -ne "default")
     {
-        throw "Could not find JDK $jdkVersion $jdkArchitecture, please make sure the selected JDK is installed properly"
+        $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
+        if (!$jdkPath) 
+        {
+            throw "Could not find JDK $jdkVersion $jdkArchitecture, please make sure the selected JDK is installed properly"
```

Comment:
For any host/warning/error user facing strings, use the cmdlet Get-LocalizedString. If you're using it with a throw, you'll need to wrap it in parenthese so PowerShell can interpret the command correctly. E.g.:
throw (Get-LocalizedString -Key 'Some {0} fancy format string goes {1} here.' -ArgumentList 'really', 'right')

There are some caveats to using the cmdlet due to the unit tests we have that generate the resx content:
1) Don't use here-strings
2) You must specify the command on one line... this can likely be relaxed in the future, but that's how it is right now due to the regex in the unit test.
3) You can use single quoted strings or double quoted strings, but you can't use run-time evaluated expressions within the double quoted strings. This is b/c we need to give the string to the localizers, so we can't take a dependency on run-time info. So you can't use any of the following for the format string: "1 + 2 = $(1 + 2)", "someVariable = $someVariable", "some variable = ${some variable}"

Lastly, the ArgumentList parameter is optional.